oasis/cloud-migration-tool.md

```diff
@@ -0,0 +1,228 @@
+  --source.cacert=tls-ca.crt \
+  --source.keyfile=client-auth.keyfile \
+  --source.endpoint=$COORDINATOR_ENDPOINT \
+  --source.jwtSecret=clusterSecret \
+  --oasis.api-key=$OASIS_API_KEY \
+  --oasis.api-secret=$OASIS_API_SECRET \
+  --oasis.deployment-id=$OASIS_DEPLOYMENT_ID \
+  --agent.listen-host=$MG_HOST
+```
+
+### TLS server certificates
+
+The migration agent HTTPS server uses TLS certificate pairs to ensure a secure
+connection between your local cluster and the Oasis platform.
+If you do not provide them, the migration tool creates self-signed certificates.
+If you wish to provide TLS certificates, use the `arangodb` tool to convert them in
+a suitable format for the migration tool.
+See a detailed guide on how to [create a new certificate/keyfile pair](../programs-starter-security.html).
+Make sure to specify your publicly available host name, `$MG_HOST` when creating
+the keyfile. 
+
+When starting the migration, specify the generated files on the command line:
+
+```bash
+--agent.cacert=tls.crt --agent.keyfile=tls.keyfile --agent.client-auth-cacert=client-auth-ca.crt --agent.client-auth-keyfile=client-auth.keyfile
+```
+
+### How long does it take?
+
+The total time required to complete the migration depends on how much data you
+have and how often write operations are executed during the process.
+
+You can also track the progress by checking the **Migration status** section of
+your target deployment in Oasis dashboard.
+
+![Oasis Cloud Migration Progress](images/oasis-migration-agent.png)
+
+### Getting the current status
+
+To print the current status of the migration, run the following command:
+
+```bash
+./arangosync-migration status --watch \
+  --oasis.api-key=$OASIS_API_KEY \
+  --oasis.api-secret=$OASIS_API_SECRET \
+  --oasis.deployment-id=$OASIS_DEPLOYMENT_ID
+```
+
+You can also add the `--watch` option to start monitoring the status in real-time.
+
+### Stopping the migration process
+
+The `arangosync-migration stop` command stops the migration and terminates
+the migration agent process.
+
+If replication is running normally, the command waits until all shards are
+in sync. The local cluster is then switched into read-only mode.
+After all shards are in-sync and the migration stopped, the target deployment
+is switched into normal mode (read/write) while the source cluster stays in
+read-only mode. 
+
+```bash
+./arangosync-migration stop \
+  --oasis.api-key=$OASIS_API_KEY \
+  --oasis.api-secret=$OASIS_API_SECRET \
+  --oasis.deployment-id=$OASIS_DEPLOYMENT_ID
+```
+
+An optional `--abort` option is supported. If specified, the `stop` command 
+will not check anymore if both deployments are in-sync and stops all
+migration-related processes as soon as possible.
+
+### Switching the local cluster to read-only mode
+
+The `arangosync-migration set-server-mode` command allows switching [read-only mode](../http/administration-and-monitoring.html#update-whether-or-not-a-server-is-in-read-only-mode)
```

Comment:
@Robert-Stam 
I see two scenarios where it can be useful (both described at the end of the doc):
1) user can switch the source deployment into read-only mode before calling `stop` - it will allow the synchronisation to catch up. `stop` does the same before actually stopping but has some timeouts (which may be exceeded if arangodb is experiencing a lot of writes for a long period).
2) In future, the `stop` command will switch the source deployment into 'read-only' mode by default - it is currently not implemented. But user may want to switch it back to default read/write for his own reasons.